src/game_strings.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ // Headers
+#include <regex>
+#include "game_strings.h"
+#include "game_variables.h"
+#include "output.h"
+
+Game_Strings::Game_Strings()
+{}
+
+void Game_Strings::WarnGet(int id) const {
+	Output::Debug("Invalid read strvar[{}]!", id);
+	--_warnings;
+}
+
+Game_Strings::Str_t Game_Strings::Asg(Str_Params params, Str_t string) {
+	return Set(params, string);
+}
+
+Game_Strings::Str_t Game_Strings::Cat(Str_Params params, Str_t string) {
+	if (!ResizeWithId(params.string_id)) return "";
+
+	Str_t s = Get(params.string_id);
+	std::string op_string = static_cast<std::string>(s);
+	op_string.append(static_cast<std::string>(string));
+	Set(params, static_cast<Str_t>(op_string));
+	return s;
+}
+
+int Game_Strings::ToNum(Str_Params params, int var_id) {
+	if (!ResizeWithId(params.string_id)) return -1;
+	std::string str = static_cast<std::string>(Get(params.string_id));
+
+	int num;
+	if (params.hex)
+		num = std::stoi(str, 0, 16);
+	else
+		num = std::stoi(str);
+
+	Main_Data::game_variables->Set(var_id, num);
+	return num;
+}
+
+int Game_Strings::GetLen(Str_Params params, int var_id) {
+	if (!ResizeWithId(params.string_id)) return -1;
+
+	int len = static_cast<std::string>(Get(params.string_id)).length();
+	Main_Data::game_variables->Set(var_id, len);
+	return len;
+}
```

Comment:
When testing the string `My name is Đức`:
Assigned to a string directly, length is 14.
When read from a file specified as utf8, length is 14.
When read from a file specified as ansi, length is 17.
![image](https://github.com/EasyRPG/Player/assets/3581299/98f38349-107e-4dad-8e9c-f57e69bb4812)

Reply:
With the PR as it is, the player does not behave the same way.
With files read as utf8, we are saying its length is 17
With files read as ansi, we are saying its length is 17
And when assigned directly to a string, we are saying its length is 15... I _believe_ this is because `tpc` doesn't allow the `ứ` character (either due to my computer's locale or tpc itself, no idea).
![image](https://github.com/EasyRPG/Player/assets/3581299/e1ee1c33-0b84-45e8-b723-be59470d9dbf)
![image](https://github.com/EasyRPG/Player/assets/3581299/b712460e-c69d-4fc1-9c97-b6777f3a23e2)

Reply:
Looks like BingShan recently released an update to `tpc` on his private discord that does not encode the text-editor as shift-jis, so I may be able to test this a little more comprehensively... I'll check this out.